sematic/api/endpoints/runs.py

```diff
@@ -280,25 +285,41 @@ def get_logs_endpoint(user: Optional[User], run_id: str) -> flask.Response:
     return flask.jsonify(payload)
 
 
-def _get_run_if_modified(
-    run_id: str, future_state: FutureState, jobs: List[ExternalJob]
-) -> Optional[Run]:
-    """Returns the updated run, if it has changed based on external job status, or None
-    if it hasn't.
+def _get_run_and_jobs_if_modified(
+    run_id: str, future_state: FutureState, jobs: List[Job]
+) -> Tuple[Optional[Run], List[Job]]:
+    """Returns updated run and jobs.
+
+    Parameters
+    ----------
+    run_id:
+        The id of the run to get updates for.
+    future_state:
+        The state of the run before checking against the jobs.
+    jobs:
+        The jobs to update.
+
+    Returns
+    -------
+    A tuple where the first element is the updated run, if the run has been modified.
+    If the run has not been modified, the first element will be None. The second element
+    is a list of the jobs after update.
     """
-    new_future_state, new_external_jobs = update_run_status(future_state, jobs)
+    new_future_state, updated_jobs = update_run_status(future_state, jobs)
     run = None
 
     # we standardize to tuples, but sometimes we get lists
-    if tuple(new_external_jobs) != tuple(jobs):
+    if tuple(updated_jobs) != tuple(jobs):
         run = get_run(run_id)
-        run.external_jobs = new_external_jobs
-        run.external_exception_metadata = new_external_jobs[-1].get_exception_metadata()
-        logger.info("Updating run's external jobs: %s", new_external_jobs)
+        if new_future_state not in (FutureState.CANCELED, FutureState.RESOLVED):
```

Comment:
Please consider defining this function, similar to the existing `is_terminal()`:
```suggestion
        if not new_future_state.is_failure():
```

Reply:
Addressed. Changed to:
```python
        # We want exception metadata if it is present while the run is still
        # active, or if it has ended in a failure state. If the run has ended
        # in a non-failure state, we don't care about exception metadata from
        # the job, because it may just be something non-standard in job cleanup
        # that doesn't actually impact the outcome of the run.
        if (
            new_future_state not in FutureState.terminal_states()
            or new_future_state in FutureState.failure_terminal_states()
        ):
```